### ISSUE TITLE
[bug] proxyNCA loss also has parameters that needs to optimized

### ISSUE THREAD

**Issue by jdhao** [tjddus9597/Proxy-Anchor-CVPR2020#25]
proxyNCA loss is also used [here](https://github.com/tjddus9597/Proxy-Anchor-CVPR2020/blob/82fd1e23b08b23331dadf2dbb4ba396201d448e3/code/train.py#L225) to compare with proxy anchor loss.

According to [doc of proxyNCA loss](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#proxyncaloss), this loss have parameters that need optimization:

> This loss requires an optimizer. You need to create an optimizer and pass this loss's parameters to that optimizer.

However, in your code, you only optimize the parameters of the proposed proxy anchor loss, as shown [here](https://github.com/tjddus9597/Proxy-Anchor-CVPR2020/blob/82fd1e23b08b23331dadf2dbb4ba396201d448e3/code/train.py#L241-L242).

Are there any hidden reasons untold or just an oversight?

**Comment by sung-yeon-kim** [comment]
Thanks for pointing out a helpful issue.
When releasing this code, we replaced the original implementation code with pytorch-metric-learning library for simplicity.
At that time, it seems that I accidentally omitted the part of optimizing parameters (i.e., proxy) of proxy-NCA.
I note that the parameters of proxy-NCA are optimized when we conducted experimentation of proxy-NCA (e.g., Fig. 1). 
Thanks again, and I will correct that code soon. :D

**Comment by jdhao** [comment]
@tjddus9597  Thanks for the prompt reply!

BTW, can you review my PR https://github.com/tjddus9597/Proxy-Anchor-CVPR2020/pull/24 here? 😃

**Comment by sung-yeon-kim** [comment]
I've been very busy lately, so I haven't been able to check the repository. I am sorry for that. 
I will check it right away and reflect that PR. :)
